### PR TITLE
Use AI-Mask

### PR DESCRIPTION
**Pull request by pacwoodson**
Very basic modifications to replace ollama by [AI-Mask extension](https://github.com/pacoccino/ai-mask)

This is a WIP, just to validate it works. 

Main issue is: I had to move back the code that was in the web worker to the main app thread as they are isolated and can't communicate with extensions (`chrome` object undefined)
The embedding phase is now blocking the UI, but that's temporary

TODO:

- [x] Make it work
- [ ] Move embedding calculations from main thread to AI-Mask
- [ ] Create proper langchain libs for AI-Mask
- [ ] Move back voy into a worker ?

### CODE REVIEW

**Comment by jacoblee93** [comment]
Will check this out this weekend! Thank you for the PR :)

**Comment by jacoblee93** [comment]
This looks really cool but yeah I think moving out of the worker/blocking on things like embeddings wouldn't be worth it yet - I'm looking into raw WebLLM support first and then will see about a protocol for a Chrome extension. I agree it's a much better form-factor.

CC @jasonmayes who had a similar idea!

File: lib/web-llm.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import {
```
Comment (jacoblee93) [inline lib/web-llm.tsx:1]: Would love to add this directly to LangChain.js - will open a PR to this repo first with streaming support and see how that goes.

**Comment by pacwoodson** [comment]
Closing in favour of https://github.com/jacoblee93/fully-local-pdf-chatbot/pull/19

**Pull request closed without merging**